coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CalciteTrinoUDFOperatorTransformers.java

```diff
@@ -0,0 +1,220 @@
+
+import com.google.common.collect.ImmutableList;
+
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+
+import com.linkedin.coral.com.google.common.base.CaseFormat;
+import com.linkedin.coral.com.google.common.base.Converter;
+import com.linkedin.coral.com.google.common.collect.ImmutableMultimap;
+import com.linkedin.coral.common.functions.Function;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+import com.linkedin.coral.common.transformers.SqlCallTransformers;
+import com.linkedin.coral.common.transformers.StandardUDFOperatorTransformer;
+import com.linkedin.coral.hive.hive2rel.functions.HiveRLikeOperator;
+import com.linkedin.coral.hive.hive2rel.functions.StaticHiveFunctionRegistry;
+import com.linkedin.coral.trino.rel2trino.functions.TrinoElementAtFunction;
+import com.linkedin.coral.trino.rel2trino.transfomers.MapStructAccessOperatorTransformer;
+
+
+/**
+ * This class defines a list of SqlCallTransformers which convert UDF operator from Calcite to Trino
+ * on SqlNode layer
+ */
+public class CalciteTrinoUDFOperatorTransformers extends SqlCallTransformers {
+  private static final StaticHiveFunctionRegistry HIVE_REGISTRY = new StaticHiveFunctionRegistry();
+  private static CalciteTrinoUDFOperatorTransformers instance;
+
+  private CalciteTrinoUDFOperatorTransformers() {
+  }
+
+  private CalciteTrinoUDFOperatorTransformers(List<SqlCallTransformer> operatorTransformers) {
+    super(ImmutableList.copyOf(operatorTransformers));
+  }
+
+  public static CalciteTrinoUDFOperatorTransformers getInstance() {
+    if (instance == null) {
+      return new CalciteTrinoUDFOperatorTransformers(initializeOperatorTransformers());
+    }
+    return instance;
+  }
+
+  private static List<SqlCallTransformer> initializeOperatorTransformers() {
+    List<SqlCallTransformer> operatorTransformers = new ArrayList<>();
+    // conditional functions
+    operatorTransformers.add(createOperatorTransformer(hiveToCalciteOp("nvl"), 2, "coalesce"));
+    // Array and map functions
+    operatorTransformers.add(createOperatorTransformer(SqlStdOperatorTable.ITEM, 2, TrinoElementAtFunction.INSTANCE));
+
+    // Math Functions
+    operatorTransformers.add(createOperatorTransformer(SqlStdOperatorTable.RAND, 0, "RANDOM"));
+    operatorTransformers.add(createOperatorTransformer(SqlStdOperatorTable.RAND, 1, "RANDOM", "[]", null));
+    operatorTransformers.add(createOperatorTransformer(SqlStdOperatorTable.RAND_INTEGER, 1, "RANDOM"));
+    operatorTransformers
+        .add(createOperatorTransformer(SqlStdOperatorTable.RAND_INTEGER, 2, "RANDOM", "[{\"input\":2}]", null));
+    operatorTransformers.add(createOperatorTransformer(SqlStdOperatorTable.TRUNCATE, 2, "TRUNCATE",
+        "[{\"op\":\"*\",\"operands\":[{\"input\":1},{\"op\":\"^\",\"operands\":[{\"value\":10},{\"input\":2}]}]}]",
+        "{\"op\":\"/\",\"operands\":[{\"input\":0},{\"op\":\"^\",\"operands\":[{\"value\":10},{\"input\":2}]}]}"));
```

Comment:
If we have this code added here, should not it be deleted from `CalciteTrinoUDFMap`?

Reply:
+1, `CalciteTrinoUDFMap` can be removed.

Reply:
Good point, `CalciteTrinoUDFMap` is removed in the PR